packages/react/src/contexts/IonLifeCycleContext.tsx

```diff
@@ -63,8 +88,64 @@ export const DefaultIonLifeCycleContext = class implements IonLifeCycleContextIn
     }
   }
 
+  teardownCallback(callback: LifeCycleCallback, callbacks: any[]) {
+    // Find any destructors that have been registered for the callback
+    const matches = callbacks.filter((x) => x.id === callback.id);
+    if (matches.length !== 0) {
+      // Execute the destructor for each matching item
+      matches.forEach((match) => {
+        if (match && typeof match.destructor === 'function') {
+          match.destructor();
+        }
+      });
+      // Remove all matching items from the array
+      callbacks = callbacks.filter((x) => x.id !== callback.id);
+    }
+  }
+
+  /**
+   * Tears down the user-provided ionViewWillEnter lifecycle callback.
+   * This is the same behavior as React's useEffect hook. The callback
+   * is invoked when the component is unmounted.
+   */
+  cleanupIonViewWillEnter(callback: LifeCycleCallback) {
+    this.teardownCallback(callback, this.ionViewWillEnterDestructorCallbacks);
+  }
+
+  /**
+   * Tears down the user-provided ionViewDidEnter lifecycle callback.
+   * This is the same behavior as React's useEffect hook. The callback
+   * is invoked when the component is unmounted.
+   */
+  cleanupIonViewDidEnter(callback: LifeCycleCallback) {
+    this.teardownCallback(callback, this.ionViewDidEnterDestructorCallbacks);
+  }
+
+  /**
+   * Tears down the user-provided ionViewWillLeave lifecycle callback.
+   * This is the same behavior as React's useEffect hook. The callback
+   * is invoked when the component is unmounted.
+   */
+  cleanupIonViewWillLeave(callback: LifeCycleCallback) {
+    this.teardownCallback(callback, this.ionViewWillLeaveDestructorCallbacks);
+  }
+
+  /**
+   * Tears down the user-provided ionViewDidLeave lifecycle callback.
+   * This is the same behavior as React's useEffect hook. The callback
+   * is invoked when the component is unmounted.
+   */
+  cleanupIonViewDidLeave(callback: LifeCycleCallback) {
+    this.teardownCallback(callback, this.ionViewDidLeaveDestructorCallbacks);
+  }
+
   ionViewWillEnter() {
-    this.ionViewWillEnterCallbacks.forEach((cb) => cb());
+    this.ionViewWillEnterCallbacks.forEach((cb) => {
+      const destructor = cb();
+      if (cb.id) {
```

Comment:
Wouldn't it be better to create the variable, `destructor` inside the if statement? That way there isn't an unused variable if `cb.id` doesn't exist.

Reply:
Yes it would!

Reply:
Following up on why I did this - took me a second to recall. 

The callback may not have an `id`, but we still want to invoke the callback regardless. Otherwise the developers implementation within the lifecycle hook will never fire. 

We only want to store the destructor callback if the callback has an `id` - which is assigned from the lifecycle hooks. 

This was intentional.